packages/expo-notifications/ios/EXNotifications/Notifications/Scheduling/EXNotificationSchedulerModule.m

```diff
@@ -0,0 +1,158 @@
+
+#import <UserNotifications/UserNotifications.h>
+
+static NSString * const notificationTriggerTypeKey = @"type";
+static NSString * const notificationTriggerRepeatsKey = @"repeats";
+
+static NSString * const intervalNotificationTriggerType = @"interval";
+static NSString * const intervalNotificationTriggerIntervalKey = @"value";
+
+static NSString * const dateNotificationTriggerType = @"date";
+static NSString * const dateNotificationTriggerTimestampKey = @"value";
+
+static NSString * const calendarNotificationTriggerType = @"calendar";
+static NSString * const calendarNotificationTriggerComponentsKey = @"value";
+static NSString * const calendarNotificationTriggerTimezoneKey = @"timezone";
+
+
+
+@interface EXNotificationSchedulerModule ()
+
+@property (nonatomic, weak) id<EXNotificationBuilder> builder;
+
+@end
+
+@implementation EXNotificationSchedulerModule
+
+UM_EXPORT_MODULE(ExpoNotificationScheduler);
+
+- (void)setModuleRegistry:(UMModuleRegistry *)moduleRegistry
+{
+  _builder = [moduleRegistry getModuleImplementingProtocol:@protocol(EXNotificationBuilder)];
+}
+
+# pragma mark - Exported methods
+
+UM_EXPORT_METHOD_AS(getAllScheduledNotificationsAsync,
+                    getAllScheduledNotifications:(UMPromiseResolveBlock)resolve reject:(UMPromiseRejectBlock)reject
+                    )
+{
+  [[UNUserNotificationCenter currentNotificationCenter] getPendingNotificationRequestsWithCompletionHandler:^(NSArray<UNNotificationRequest *> * _Nonnull requests) {
+    NSMutableArray *serializedRequests = [NSMutableArray new];
+    for (UNNotificationRequest *request in requests) {
+      [serializedRequests addObject:[EXNotificationSerializer serializedNotificationRequest:request]];
+    }
+    resolve(serializedRequests);
+  }];
+}
+
+UM_EXPORT_METHOD_AS(scheduleNotificationAsync,
+                     scheduleNotification:(NSString *)identifier notificationSpec:(NSDictionary *)notificationSpec triggerSpec:(NSDictionary *)triggerSpec resolve:(UMPromiseResolveBlock)resolve rejecting:(UMPromiseRejectBlock)reject)
+{
+  @try {
+    UNNotificationContent *content = [_builder notificationContentFromRequest:notificationSpec];
+    UNNotificationRequest *request = [UNNotificationRequest requestWithIdentifier:identifier content:content trigger:[self triggerFromParams:triggerSpec]];
+    [[UNUserNotificationCenter currentNotificationCenter] addNotificationRequest:request withCompletionHandler:^(NSError * _Nullable error) {
+      if (error) {
+        NSString *message = [NSString stringWithFormat:@"Failed to schedule notification. %@", error];
+        reject(@"ERR_FAILED_TO_SCHEDULE", message, error);
+      } else {
+        resolve(nil);
+      }
+    }];
+  } @catch (NSException *exception) {
+    NSString *message = [NSString stringWithFormat:@"Failed to schedule notification. %@", exception];
+    reject(@"ERR_FAILED_TO_SCHEDULE", message, nil);
+  }
+}
+
+UM_EXPORT_METHOD_AS(cancelScheduledNotificationAsync,
```

Comment:
I am wondering if it might be worth providing a `cancelAllScheduledNotificationsAsync` method too, which calls the underlying `removeAllPendingNotificationRequests` method on UNUserNotificationCenter. I guess you didn't include it because you can achieve the same functionality by `getAllScheduledNotificationsAsync` --> cancel one at a time. But since it's part of our current notifications API, and it's a separate native method, and it's an easy catch-all for developers to get to a clean slate, I think it might be worth including.

Reply:
Added! 👍